### PR TITLE
Add contributing docs

### PR DESCRIPTION
**Pull request by prescod**
Add CONTRIBUTING.md and command line validation for `proof-of-concept/test_pof.sh`

### CODE REVIEW

**Comment by antocuni** [comment]
Thank you, it looks like a good idea!
I'm +1 to merge this, but I'll wait some time so that others can review it as well.

**Review by hodgestar — APPROVED** [review]
Thanks! Looks good to me too.

File: proof-of-concept/test_pof.sh [hunk 1]
```diff
@@ -105,6 +105,24 @@ setup_py_install() {
 
 # ======== main code =======
 
+# validate arguments
+if [[ "$#" -lt 1  || ( "$#" -lt 2  && "$1" != "clean") ]]; then
+  echo "Usage: $0 COMMAND [TARGET_ABI]" >&2
+  echo "Commands:" >&2
+  echo "  wheel TARGET_ABI: build a wheel, install and test"
+  echo "  setup_py_install TARGET_ABI: install poc using 'setup.py install' & run tests" >&2
+  echo "  clean: clean build artifacts" >&2
```
Comment (hodgestar) [inline proof-of-concept/test_pof.sh:114]: I added another command in #143, namely `setup_py_build_ext_inplace`, but I can add the documentation for that after this PR is merged.

**Comment by hodgestar** [comment]
Since we have two +1s, I will merge this and we can always fix things if needed.

**Comment by hodgestar** [comment]
@prescod Thanks!

**Pull request merged**